mkosi.images/system/mkosi.conf.d/10-arch/mkosi.build.chroot

```diff
@@ -24,13 +24,13 @@ ln --symbolic . "pkg/$ID/src"
 # shellcheck source=/dev/null
 . /etc/makepkg.conf
 
-CFLAGS="-O${OPTIMIZATION:-0} -Wp,-U_FORTIFY_SOURCE"
+CFLAGS="$CFLAGS -O${OPTIMIZATION:-0} -Wp,-U_FORTIFY_SOURCE"
```

Comment:
I think if CFLAGS supplied externally took precedence over CFLAGS provided by mkosi it would be possible to easily set FORTIFY_SOURCE. It would be off by default to avoid interfering with ASan/UBSan and other things and can be overridden. Would maybe something like
```
MKOSI_CFLAGS='-O${OPTIMIZATION:-0} -Wp,-U_FORTIFY_SOURCE'
...
if ((LLVM)); then
    MKOSI_CFLAGS+=" -shared-libasan -fno-sanitize=function"
fi

CFLAGS="$MKOSI_CFLAGS $CFLAGS"
```
work?

Reply:
Reworked it like you suggested.